### ISSUE TITLE
Ability to change port - feature request

### ISSUE THREAD

**Issue by ghost** [qTox/qTox#2581]
I think that qTox should allow you to change which port it is running on. This should be configurable in the settings.

**Comment by ElLamparto** [comment]
Agreed 100%. This is very important. It would increase network efficiency and quality of connections.
At work I have some ports routed to my machine, but not the Tox one.
Also while connecting over a VPN, the Tox port is always taken by someone and can not be opened.

**Comment by tux3** [comment]
That's not possible with the current toxcore API, and it's unlikely to be added in the short term. You should try opening an issue on the toxcore repo, but in the meantime there's not much qTox can do about it.

**Comment by ghost** [comment]
Ok, well I have suggested it [here](https://github.com/irungentoo/toxcore/issues/1481).

**Comment by iphydf** [comment]
As commented on the linked issue, it is possible, unless I misunderstand the idea here.

**Comment by tux3** [comment]
Turns out this is absolutely possible.

**Comment by ElLamparto** [comment]
Sounds great! But how to do it? IS there a command line parameter or I have to modify and recompile Tox ??

**Comment by zetok** [comment]
@ElLamparto You would have to modify & compile qTox

**Comment by ElLamparto** [comment]
@zetok, ok, so let's keep it as a feature request.
By the way, I miss explanation on Tox usage of ports. UDP, TCP, port ranges etc. Perhaps not on the main web page, but it should be available somewhere for advanced users.
